config/routes.rb

```diff
@@ -2,60 +2,81 @@
 
 # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 Rails.application.routes.draw do
-  resources :app_settings, only: %i[update edit]
-  resources :operating_systems, path: '/vms/requests/operating_systems', except: :show
+  # You can specify what Rails should route '/' to with the root method
+  # https://guides.rubyonrails.org/routing.html#using-root
+  root 'landing#index'
 
-  resources :request_templates, path: '/vms/request_templates', except: :show
-  patch '/vms/requests/reject', to: 'requests#reject', as: 'reject'
-  resources :requests, path: '/vms/requests' do
-    post :push_to_git, on: :member
-  end
+  # '/dashboard'
+  get '/dashboard' => 'dashboard#index', as: :dashboard
+  # '/projects/...'
+  resources :projects, only: %i[index show new edit create update]
+  # '/app_settings/...'
+  resources :app_settings, only: %i[update edit]
+  # '/hosts/...', '/servers/...'
+  resources :hosts, :servers
+  get '/hosts/:id' => 'hosts#show', constraints: { id: /.*/ }
```

Comment:
Is this line not be included in the resources command?

Reply:
I believe the original intent of this line was the `, constraints: { id: /.*/ }`, which allows arbitrary strings as ids. Good questions why this should only be possible for the `show` actions of hosts and not the others. Should URLs for hosts refer to host objects with names? Or can the usual numerical ids be ussed?

Reply:
I think usual numerical ids are completly fine.

Reply:
I agree that numerical ids would actually be preferable to using names in URLs. However, the current system relies on VM names (e.g. [https://hart-dev.epic-hpi.de/vms/vm/many-responsible](https://hart-dev.epic-hpi.de/vms/vm/many-responsible)), presumably because the vSphere API does not return ids for VMs?

Reply:
I agree that numerical ids would actually be preferable to using names in URLs. However, the current system relies on VM names (e.g. [https://hart-dev.epic-hpi.de/vms/vm/many-responsible](https://hart-dev.epic-hpi.de/vms/vm/many-responsible)) (presumably because the vSphere API does not return ids for VMs?). The use of vm names in URLs in the current system is the explanation for the separate `constraints: { id: /.*/ }` code, allowing this, where usually only numerical ids would be allowed.

Reply:
Related ticket: https://github.com/hpi-swt2/vm-portal/issues/480